### PR TITLE
Support pluggable custodian monitor

### PR DESCRIPTION
**Pull request by jaydoane**
## Overview

Enable build time configurable monitor for custodian and remove custom
sensu events.

## Testing recommendations

```
make eunit apps=custodian
```

## Checklist

- [x] Code is written and works correctly
- [x] Changes are covered by tests
- [ ] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [ ] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

File: src/custodian/src/custodian_monitor.erl [hunk 1]
```diff
@@ -10,20 +10,16 @@
...
-                  mem3
-                 ]},
-  {mod, { custodian_app, []}},
-  {env, []}
- ]}.
+-module(custodian_monitor).
+
+
+-callback send_missing_db_alert(DbName :: binary()) ->
+    Output :: string().
```
Comment (rnewson) [inline src/custodian/src/custodian_monitor.erl:17]: I'd encode the fact that we don't use the result of these functions, we call them only for their implementation. so perhaps insist on `false` atom in all cases or say `any()` and document that the result of the function is ignored.

File: src/custodian/src/custodian_noop_monitor.erl [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+-export([
+    send_missing_db_alert/1,
+    clear_missing_dbs_alert/0,
+    send_event/1
+]).
+
+
+send_missing_db_alert(_DbName) ->
+    "no-op".
```
Comment (rnewson) [inline src/custodian/src/custodian_noop_monitor.erl:27]: adjust these to `false`.

File: src/custodian/src/custodian_server.erl [hunk 3]
```diff
@@ -143,52 +143,7 @@ handle_db_event(_DbName, _Event, _St) ->
     {ok, nil}.
 
 check_shards() ->
-    [send_sensu_event(Item) || Item <- custodian:summary()].
```
Comment (rnewson) [inline src/custodian/src/custodian_server.erl:146]: this is a cut too far imo. we still want to log the descriptive error message to couch_log. the pluggable module allows a sysadmin to send that message (or one of its own invention) to some other system.

**Review by rnewson — CHANGES_REQUESTED** [review]
Almost :)

File: src/custodian/src/custodian_monitor.erl (hunk 1 above)
Comment (jaydoane) [reply src/custodian/src/custodian_monitor.erl]: Done in fixup

File: src/custodian/src/custodian_noop_monitor.erl (hunk 2 above)
Comment (jaydoane) [reply src/custodian/src/custodian_noop_monitor.erl]: Done

File: src/custodian/src/custodian_server.erl (hunk 3 above)
Comment (jaydoane) [reply src/custodian/src/custodian_server.erl:146]: Hah, I didn't even notice that logging cleverly interleaved in the case statement. In any case, I've returned them, and changed the `send_event/3` callback to DRY things up a bit.

File: src/custodian/src/custodian_server.erl [hunk 4]
```diff
@@ -143,7 +143,49 @@ handle_db_event(_DbName, _Event, _St) ->
...
+        0 ->
+            "--ok";
+        1 ->
+            couch_log:critical("~s", [Description]),
+            "--critical";
+        _ ->
+            couch_log:warning("~s", [Description]),
+            "--warning"
+    end,
+    ?CUSTODIAN_MONITOR:send_event(Name, Level, Description).
```
Comment (rnewson) [inline src/custodian/src/custodian_server.erl:162]: think we should send `Count` itself, the transformation to `Level` is sensu-specific. 😬

**Review by rnewson — APPROVED** [review]
nice! please squash before merging!

**Pull request merged**